src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -292,7 +292,7 @@ class InstanceOpFactoryImpl(
         }(collection.breakOut)
 
         val containerNameToTaskId: Map[String, Task.Id] = oldToNewTaskIds.values.map {
-          case taskId @ Task.ResidentTaskId(_, Some(containerName), _) => containerName -> taskId
+          case taskId @ Task.TaskIdWithIncarnation(_, Some(containerName), _) => containerName -> taskId
```

Comment:
I'd also introduce it for ephemeral instances as well

```scala
val taskId = scheduledInstance.tasksMap.headOption match {
  case Some((taskId, _)) => Task.Id.withIncarnationCount(taskId)
  case None => Task.Id.forInstanceId(scheduledInstance.instanceId)
}
```

This way #6616 would not have to introduce that change.

Reply:
Also, what do you think about

```scala
val taskId = scheduledInstance.tasksMap.headOption match {
  case Some((taskId, _)) => taskId.withIncreasedIncarnation()
  case None => Task.Id.forInstanceId(scheduledInstance.instanceId)
}
```